Project/Assets/ML-Agents/Examples/Walker/Scripts/WalkerAgent.cs

```diff
@@ -1,16 +1,36 @@
+using MLAgentsExamples;
 using UnityEngine;
 using Unity.MLAgents;
 using Unity.MLAgentsExamples;
 using Unity.MLAgents.Sensors;
+using BodyPart = Unity.MLAgentsExamples.BodyPart;
 
 public class WalkerAgent : Agent
 {
+    [Header("Walking Speed")]
+    [Space(10)]
     [Header("Specific to Walker")]
+    public float maximumWalkingSpeed = 999; //The max walk velocity magnitude an agent will be rewarded for
+    Vector3 m_WalkDir; //Direction to the target
+    Quaternion m_WalkDirLookRot; //Will hold the rotation to our target
+    
+    [Space(10)]
+    [Header("Orientation Cube")]
+    [Space(10)]
+    //This will be used as a stable observation platform for the ragdoll to use.
+    GameObject m_OrientationCube;
+    
     [Header("Target To Walk Towards")]
     [Space(10)]
+    public float targetSpawnRadius;
     public Transform target;
+    public Transform ground;
+    public bool detectTargets;
+    public bool targetIsStatic;
+    public bool respawnTargetWhenTouched;
```

Comment:
I think it should be the target that knows what behavior to have when touched, not the Agent.

From the videos you showed, some Agents can have the same target, if they have different settings, the target's behavior will be ambiguous.